dspace-api/src/main/java/org/dspace/content/DSpaceObject.java

```diff
@@ -192,4 +202,70 @@ public void clearModified() {
     protected void setModified() {
         this.modified = true;
     }
+
+    public String toString() {
+        String ident = getHandle();
+        if (ident == null)
+            ident = "" + getID();
+        return Constants.typeText[getType()] + "." + ident;
+    }
+
+    /**
+     * expects a handle or a two part string as parameter: TYPE.IDENT,
+     * where TYPE is a DSpaceObject type and
+     * IDENT is a UUID, handle,  group name, or Eperson's netid
+     *
+     * returns DSpaceObject that corresponds to string or null
+     * @param str    must have format TYPE.ID
+     * @return
+     */
+    public static DSpaceObject fromString(Context c, String str) throws SQLException {
+        if (str == null) {
+            return null;
+        }
+        int doti = str.indexOf('.');
+        HandleService handleService = HandleServiceFactory.getInstance().getHandleService();
+        if (doti == -1) {
+            // no '.' ==> assume its a handle
+            return handleService.resolveToObject(c, str);
+        }
+
+        if (doti > 0 && doti < str.length()-1) {
```

Comment:
Why do we still need the "type", could you not just use the UUID, the UUID is unique over all DSpaceObjects, so type is no longer needed.